packages/compiler/src/myst.ts

```diff
@@ -0,0 +1,41 @@
+import { micromark } from "micromark";
+
+export function parsePanels (contents) {
+   const panelDelimiterRegex = /\n\-{3,}\n/;
+  const headerDelimiterRegex = /\n\^{3,}\n/;
+  const footerDelimiterRegex = /\n\+{3,}\n/;
+
+  // first retrieve cards
+  const cards = contents.split(panelDelimiterRegex);
+  let splitCards = [];
+
+  // retrieve header and footer if exists
+  for (const card of cards) {
+    // default empty string
+    let header = "";
+    let body = card;
+    let footer = "";
+    let contents;
+
+    if (headerDelimiterRegex.test(body)) {
+      contents = body.split(headerDelimiterRegex)
+      if (contents.length == 2) {
+        [header, body]  = contents
+      } else {
+        console.log("Invalid syntax for panel header.");
```

Comment:
I think we should probably throw an exception here (and in other places where there's unexpected input). e.g. https://github.com/irydium/irydium/blob/1e7c42a55bc0632ef5bf841176fd8b10bb280893/packages/compiler/src/plugins.ts#L211

At some point I'd like to make irydium gracefully degrade when things are poorly specified (maybe putting an error banner inline) but I think it's better to fail noisily rather than have undefined behaviour

Reply:
Added error throwing for invalid panel syntax.